src/DataManagement/MonthlyReport/MonthlyReportCalculator.cs

```diff
@@ -0,0 +1,135 @@
+            UserId = userId;
+        }
+
+        public async Task<MonthlyReportResult> Calculate()
+        { 
+            Result = await GatherDataAsync();
+            CalculateBiggestExpensesCategory();
+            Result.SavedUpGoals = CalculateSavedUpGoals();
+            CalculateGoalReport();
+            return Result;
+        }
+
+        private async Task<MonthlyReportResult> GatherDataAsync()
+        {
+            var today = DateTime.Today;
+            var month = new DateTime(today.Year, today.Month, 1);
+            var first = month.AddMonths(-1);
+            var last = month.AddDays(-1);
+
+            Expenses = await EntryDatabase.ReadListAsync(UserId, EntryType.Expense);
+            Income = await EntryDatabase.ReadListAsync(UserId, EntryType.Income);
+            Goals = await GoalDatabase.ReadListAsync(UserId);
+
+            var expensesTotal = Expenses.GetSum();
+            var incomeTotal = Income.GetSum();
+            var thisMonthExpenses = Expenses.GetFrom(first).GetTo(last).GetSum();
+            var thisMonthIncome = Income.GetFrom(first).GetTo(last).GetSum();
+            AllSavings = incomeTotal - expensesTotal;
+            var balance = thisMonthIncome - thisMonthExpenses;
+
+            return new MonthlyReportResult(thisMonthExpenses, thisMonthIncome, balance, first, last);
+
+        }
+
+        private void CalculateBiggestExpensesCategory()
+        {
+            var enumCount = Enum.GetValues(typeof(Importance)).Length;
+            var biggestCategory = (Importance) enumCount;
+            var sum = 0m;
+            foreach (var importance in Enum.GetValues(typeof(Importance)).Cast<Importance>())
+            {
+                var temp = 0M;
+                var expenses = Expenses.GetBy((Importance)importance);
+                foreach (var entry in expenses) //not linq since when empty category it crashes
+                {
+                    temp += entry.Amount;
+                }
+                if (temp < sum) continue;
+                sum = temp;
+                biggestCategory = (Importance)importance;
+            }
+
+            Result.BiggestCategory = biggestCategory;
+            Result.BiggestCategorySum = sum;
+            var necessarySum = Expenses.GetBy(Importance.Necessary).Sum(entry => entry.Amount);
+            Result.NecessarySum = necessarySum;
+            if (necessarySum == 0)
+            {
+                Result.HowMuchBigger = -1;
+                return;
+            }
+            Result.HowMuchBigger = ((sum - necessarySum) / necessarySum * 100);
+        }
+
+        private List<IGoal> CalculateSavedUpGoals()
+        {
+            return AllSavings >= 0 ? Goals.Where(x => x.Amount <= AllSavings).ToList() : new GoalList();
+        }
+
+        private void CalculateGoalReport()
+        {
+            var savedThisMonth = Income.GetSum() - Expenses.GetSum();
+            if (Goals.Count() < 2 || savedThisMonth <= 0)
```

Comment:
The braces after Count are unnecessary, I  think there is a property called Count so calling the method Count() is not needed